views/post.py

```diff
@@ -68,3 +68,65 @@ def create_post(post):
         conn.commit()
 
         return db_cursor.lastrowid
+
+
+def get_post(pk, query_params=None):
+    if query_params and "_expand" in query_params:
```

Comment:
Because we will need the expanded username regardless of whether the client provides "_expand" in the fetch url, I suggest we simplify this method by removing the if/else block and providing the expanded user information along with the post details whether the client explicitly asks for it or not. 

If we want to keep the if/else block, then you'll need to pass in url['query_params'] into the line where you invoke this function in json-server.py in order for the if block to execute.